aws_jobs/cyhy-data-extract.py

```diff
@@ -132,40 +134,36 @@ def query_data(collection, query, tbz_file, tbz_filename, end_of_data_collection
         end_of_data_collection.isoformat().replace(":", "").split(".")[0],
     )
     collection_file = open(json_filename, "w")
```

Comment:
It is better to use the `with open(json_filename, "w") as collection file:` idiom here, since it will automatically close the file when `collection_file` goes out of scope.

Reply:
It is my understanding that because I do not need to interact with the file, as long as I am using the with idiom it will still add the __enter__() and __exit__() methods are still called on the context manager. This is the behavior I observed after testing to verify that even without an "as f" it would still close the file.

Reply:
I'm talking about line 136 here, not line 90.

Reply:
Ah, that's what I get for not paying more attention to tagged lines.

Reply:
Updated by @mcdonnnj in c2d2f3e.